pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -178,24 +194,79 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 	return utilerrors.NewAggregate(errs)
 }
 
-func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node, recorder events.Recorder) error {
+// validateCertificateSAN makes sure that a given certificate is valid
+// at least for the SANs defined in the configuration.
+func verifyCertificateSAN(certData map[string][]byte, certName string, ipAddresses []string) error {
+	certificates, err := cert.ParseCertsPEM(certData["tls.crt"])
+	if err != nil {
+		return fmt.Errorf("could not parse TLS cert for %s: %w", certName, err)
+	}
+	// Using the first certificate to verify the IP addresses
```

Comment:
Here are a few examples where they don't even comment why they are using the zero index certificate :-)!
 
1.  https://github.com/openshift/library-go/blob/052c4104706cb9a67fc093cba11baf7980525109/pkg/assets/template.go#L50

2. https://github.com/openshift/service-ca-operator/blob/ac8a280b37d8efc765fa82f94467d30eb5674813/pkg/operator/sync_common.go#L94

3. https://github.com/openshift/service-ca-operator/blob/ac8a280b37d8efc765fa82f94467d30eb5674813/pkg/operator/sync_common.go#L128

4. https://github.com/openshift/service-ca-operator/blob/ac8a280b37d8efc765fa82f94467d30eb5674813/pkg/controller/servingcert/starter/starter.go#L98

5. https://github.com/openshift/service-ca-operator/blob/ac8a280b37d8efc765fa82f94467d30eb5674813/pkg/controller/servingcert/controller/secret_creating_controller.go#L243

6. https://github.com/openshift/cluster-kube-apiserver-operator/blob/c7b1e077d913fbe6e89d4c10312346234e31e4e4/test/e2e/user_certs_test.go#L303

7. https://github.com/openshift/service-ca-operator/blob/09bcf1403670716e32f33b51893a0ddf3cbb3aed/test/e2e/e2e_test.go#L474

:-)

Reply:
I'm afraid most of the service ca operator examples aren't applicable. What's being managed there is a CA cert (a cert used to generate other certificates rather than secure traffic), and the service ca operator performs seamless automatic rotation by cross-signing. The first cert is the current CA, and any subsequent cert is used only for validating previously issued certificates during a handover period.

For https://github.com/openshift/service-ca-operator/blob/ac8a280b37d8efc765fa82f94467d30eb5674813/pkg/controller/servingcert/starter/starter.go#L95, note the explicit check for only a single certificate.

In the library-go and ckao examples you point to, they are implicitly expecting only a single certificate. 

The comment here says 'Using the first certificate...' which begs the questions 'why the first certificate?' and 'are there other certificates and why are their ip addresses not important to check?'. I'm assuming only a single certificate is expected, which suggests following the example of library-go and ckao and just removing the comment.

Reply:
I will remove the comment. Thanks.